src/libraries/System.IO.FileSystem/tests/Directory/Move.cs

```diff
@@ -324,24 +324,14 @@ public void CaseVariantDirectoryNameWithCaseVariantPaths_CaseInsensitiveFileSyst
         }
 
         [Fact]
-        [PlatformSpecific(TestPlatforms.Windows | TestPlatforms.FreeBSD | TestPlatforms.NetBSD)]
+        [PlatformSpecific(TestPlatforms.Windows | TestPlatforms.OSX | TestPlatforms.FreeBSD | TestPlatforms.NetBSD)]
```

Comment:
probably not related but I'm wondering why the *BSD is here:
```
furt@d13:/tmp $ mkdir test
furt@d13:/tmp $ cd test
furt@d13:/tmp/test $ mkdir FOO
furt@d13:/tmp/test $ mkdir -p bar/foo
furt@d13:/tmp/test $ mv FOO bar
furt@d13:/tmp/test $ ls -al bar/
total 2
drwxr-xr-x  4 furt  wheel  4 Feb 23 22:37 .
drwxr-xr-x  3 furt  wheel  3 Feb 23 22:37 ..
drwxr-xr-x  2 furt  wheel  2 Feb 23 22:36 FOO
drwxr-xr-x  2 furt  wheel  2 Feb 23 22:36 foo
```

As far as I understand this should fail on case-insensitive filesystem but it should work on case-sensitive one.

Reply:
This is similar to your other comment:

> Once again this may be beyond intend but it seems like this should be something like IsCaseInsensitive(TestDirectory)

I agree, it would be better to capture what is assumed and express it via a `ConditionalFact` rather than hard code a list of OSes.

Should I create a ticket to clean this up?

Reply:
> I agree, it would be better to capture what is assumed and express it via a `ConditionalFact` rather than hard code a list of OSes.
> 
> Should I create a ticket to clean this up?

Sure; thank you. I don't expect we'd get to that any time soon, but it could be filed as https://github.com/dotnet/runtime/labels/up-for-grabs.